### PR TITLE
feat: Added Theme and migrate to emotion@10.x 🚀

### PR DESCRIPTION
**Pull request by priscilawebdev**
**Type:** Feat + Refactor + Chore (sorry)

**Description:**

I have added the emotion theme together with the Material Ui Theme. Since I had some problems due to the old version of emotion, I had to update the all the emotion dependencies and fix all the breaking changes. 

Of course, there's a lot to improve, but it's a start 🙃

PS: As soon as they release the fix for this issue https://github.com/emotion-js/emotion/issues/1305,  I will remove the `<{theme?: Theme}> `and add a file a file with 

```
import styled, { CreateStyled } from '@emotion/styled';

import { Theme } from './theme';

export default styled as CreateStyled<Theme>;
```

### CODE REVIEW

**Comment by anikethsaha** [comment]
Just for future tracking, referencing the migration issue https://github.com/verdaccio/ui/issues/221

**Comment by anikethsaha** [comment]
BTW, The migration to emotion 10 looking great 🤓

File: src/components/CopyToClipBoard/CopyToClipBoard.test.tsx [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
 import { copyToClipBoardUtility } from '../../utils/cli-utils';
+import { mount } from '../../utils/test-enzyme';
 
 import CopyToClipBoard from './CopyToClipBoard';
 import { CopyIcon } from './styles';
 
 jest.mock('../../utils/cli-utils');
 
 describe('<CopyToClipBoard /> component', () => {
-  let wrapper: ReactWrapper;
```
Comment (tmkn) [inline src/components/CopyToClipBoard/CopyToClipBoard.test.tsx]: Any reason why you removed the type here? This now errors again in strict mode, it was already covered :(

File: src/App/App.test.tsx [hunk 2]
```diff
@@ -34,7 +35,7 @@ jest.mock('../utils/api', () => ({
 }));
 
 describe('App', () => {
-  let wrapper: ReactWrapper<{}, AppProps, App>;
```
Comment (tmkn) [inline src/App/App.test.tsx]: Type also removed here, errors again in `strict` mode.

File: src/components/CopyToClipBoard/CopyToClipBoard.test.tsx (hunk 1 above)
Comment (juanpicado) [reply src/components/CopyToClipBoard/CopyToClipBoard.test.tsx]: It might be a mistake, I restored them.

Comment (juanpicado) [reply src/components/CopyToClipBoard/CopyToClipBoard.test.tsx]: Restored

**Comment by juanpicado** [comment]
Thanks @anikethsaha and @tmkn for the early reviews 🚀

**Comment by priscilawebdev** [comment]
thank you team 🤝💪

File: src/components/CopyToClipBoard/CopyToClipBoard.test.tsx (hunk 1 above)
Comment (priscilawebdev) [reply src/components/CopyToClipBoard/CopyToClipBoard.test.tsx]: Sorry @tmkn I was getting errors due to these types, I removed it for testing purposes and forgot to undo my changes ... good catch!

Comment (tmkn) [reply src/components/CopyToClipBoard/CopyToClipBoard.test.tsx]: @priscilawebdev no worries, I was sure it wasn't intentional, that's what code reviews are for :)

**Pull request merged**